clustermesh-apiserver/vmmanager.go

```diff
@@ -54,7 +54,7 @@ func NewVMManager(clientset k8sClient.Clientset) *VMManager {
 	if option.Config.EnableWellKnownIdentities {
 		identity.InitWellKnownIdentities(option.Config)
 	}
-	m.identityAllocator.InitIdentityAllocator(clientset, identityStore)
+	m.identityAllocator.InitIdentityAllocator(clientset, nil)
```

Comment:
Nit: looks like after this change all callers of `(*CachingIdentityAllocator). InitIdentityAllocator` set the `identityStore` argument to `nil`. So might as well remove it?

Reply:
I've opened a separate PR to complete the cleanup, to avoid increasing the scope of this one which is already fairly large: https://github.com/cilium/cilium/pull/25594